corehq/preindex/models.py

```diff
@@ -3,8 +3,12 @@
 from corehq.preindex import get_preindex_plugin
 
 
-def catch_signal(sender, **kwargs):
+def catch_signal(sender, using=None, **kwargs):
     """Function used by syncdb signal"""
+    if using != 'default':
+        # only sync for the default DB
```

Comment:
This prevents calling the preindex plugins multiple times in the sharded DB setup